src/vm/scanning.rs

```diff
@@ -107,16 +107,32 @@ pub trait RootsWorkFactory<ES: Edge>: Clone + Send + 'static {
 
     /// Create work packets to handle nodes pointed by root edges.
     ///
-    /// The work packet cannot update root edges, therefore it cannot move the objects.  This
-    /// method can only be used by GC algorithms that never moves objects, or GC algorithms that
-    /// supports object pinning.
+    /// The work packet cannot update root edges, therefore it cannot move the objects
+    /// i.e. they will be pinned for the duration of the GC
     ///
     /// This method is useful for conservative stack scanning, or VMs that cannot update some
     /// of the root edges.
     ///
     /// Arguments:
     /// * `nodes`: A vector of references to objects pointed by root edges.
-    fn create_process_node_roots_work(&mut self, nodes: Vec<ObjectReference>);
+    fn create_process_pinned_roots_work(&mut self, nodes: Vec<ObjectReference>);
+
+    /// Create work packets to handle root edges.
+    ///
+    /// The work packet must not update the edges transitively found from any of the root edges.
+    ///
+    /// Arguments:
+    /// * `edges`: A vector of edges.
+    fn create_process_tpinned_edge_roots_work(&mut self, edges: Vec<ES>);
+
+    /// Create work packets to handle nodes pointed by root edges.
+    ///
+    /// Similar to `create_process_pinned_roots_work`, this work packet won't move root objects, but also will
+    /// not move any object in their transitive closure.
```

Comment:
Emphasise the kind of roots in the first sentence.

```suggestion
    /// Create work packets to handle transitively pinning (TP) roots.
    ///
    /// Similar to `create_process_pinning_roots_work`, this work packet will not move objects in `nodes`.
    /// Unlike ``create_process_pinning_roots_work`, no objects in the transitive closure of `nodes` will be moved, either.
```